File: fwz/demo/ReaderIdcard/app/src/main/java/com/pda/nfcidcard/GuidUtils.java
package com.pda.nfcidcard;

import java.util.Locale;
import java.util.UUID;

/**
 * UUID
 *
 */
public class GuidUtils {
	private String uuid;

	private GuidUtils() {
	}
	
	public String getCurrentId() {
		if(uuid==null){
			uuid=newRandom();
		}
		return uuid;
	}
	private static GuidUtils guidUtils;
	public static GuidUtils getInstance() {
		if(guidUtils==null){
			guidUtils=new GuidUtils();
		}
		return guidUtils;
	}
	
	public String newRandom() {
		uuid=UUID.randomUUID().toString().toUpperCase(Locale.getDefault());
		return uuid;
	}

	public void clearUUID(){
		uuid = null;
	}
}

File: fwz/android/app/src/main/java/com/fwz/GuidUtils.kt
package com.fwz

import java.util.*

/**
 * GUID工具类
 */
class GuidUtils private constructor() {
    
    private var currentId: String = ""
    
    companion object {
        @Volatile
        private var INSTANCE: GuidUtils? = null
        
        fun getInstance(): GuidUtils {
            return INSTANCE ?: synchronized(this) {
                INSTANCE ?: GuidUtils().also { INSTANCE = it }
            }
        }
    }
    
    /**
     * 生成新的随机ID
     */
    fun newRandom() {
        currentId = UUID.randomUUID().toString().replace("-", "")
    }
    
    /**
     * 获取当前ID
     */
    fun getCurrentId(): String {
        return currentId
    }
}